src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php

```diff
@@ -402,6 +403,15 @@ public function get_item( $request ) {
 			return $revision;
 		}
 
+		if ( (int) $parent->ID !== (int) $revision->post_parent ) {
+			return new WP_Error(
+				'rest_post_invalid_parent',
+				/* translators: %d: A post revision id. */
+				sprintf( __( 'Parent post does not have a revision with id of "%d"' ), $request['id'] ),
```

Comment:
@ramonjd Should this message be overridden in `WP_REST_Template_Revisions_Controller`? 

https://github.com/WordPress/wordpress-develop/blob/c78e3fbce4c8af24b8feff47e6f40f9f7873a7db/src/wp-includes/rest-api/endpoints/class-wp-rest-template-revisions-controller.php#L169-L175

Similar messages there use the wording template.